jdaviz/app.py

```diff
@@ -1014,6 +1031,8 @@ def get_subsets(self, subset_name=None, spectral_only=False,
                 else:
                     all_subsets[label] = subset_region
 
+        # can this be done at the top to avoid traversing all subsets if only
+        # one is requested?
```

Comment:
This might be a question for @javerbukh .

Reply:
I also think this is relevant to @kecnry batch photometry, this could get annoying when you have a ton of subsets and are just requesting one.

Reply:
I agree, this would benefit from extra logic in the for-loop to skip parsing objects that aren't requested, but that is also out of scope for this.  Maybe create a follow-up ticket as tech debt?

Reply:
Getting the names of all subsets is helpful for the error message on line 1040 in this PR, which is why it is located here. I agree with @kecnry that it can be part of a follow-up ticket though.